apps/web/src/hooks/useApplicationInfo.ts

```diff
@@ -0,0 +1,31 @@
+import { useSearchParams } from "react-router-dom";
+
+import { Scalars } from "~/api/generated";
+
+import useRoutes from "./useRoutes";
+
+const useApplicationInfo = (userId?: Scalars["ID"]) => {
+  const paths = useRoutes();
+  const [searchParams] = useSearchParams();
+
+  let id;
+  let param;
+  // If we do not have a userId, we can let the profile redirect handle it
+  let returnRoute = userId ? paths.profile(userId) : paths.myProfile();
+
+  if (searchParams.has("applicationId")) {
+    id = searchParams.get("applicationId");
+    if (id) {
+      param = `?${searchParams.toString()}`;
```

Comment:
This isn't exactly the same behaviour as before.  Did you mean to return all the search params here?  It seems like it would match the name of the hook a little better if it just returned the applicationId or a key-value pair.  Also, if you didn't hardcode the question mark in the string you wouldn't run the risk of URL building a url with more than one question mark in the code that's using this hook.

Reply:
Honestly, I felt a little weird about returning it at all because it can easily be done just by using the ID if needed. I removed it in [update to only return application related info](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5683/commits/cff5c72f60b64e56a0d629abe79acd2986ffa9e5)